pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -299,7 +299,7 @@ def _setup_execution_config(self, config):
             updated_values["gradient_method"] = "adjoint"
         if config.use_device_gradient is None:
             updated_values["use_device_gradient"] = config.gradient_method in ("best", "adjoint")
-        if config.grad_on_execution is None:
+        if config.use_device_gradient and config.grad_on_execution is None:
```

Comment:
Thanks @andrijapau for fixing this issue! Does this change mean that we cannot set `config.use_device_gradient=True` for PL built-in gradient methods (i.e., `use_device_gradient==False`)? In other words, why do we need this additional condition to be able to updaet the `grad_on_execution` value?

Reply:
We are starting to use the execution config to control non-device behavior as well.  So now it can store gradient transforms, for example.  So if we are using a normal core gradient transform, we want to to have `grad_on_execution=False`.  

Actually, we should check both the config and the `updated_values`.

```suggestion
        if config.use_device_gradient or updated_values.get("use_device_gradient", False) and config.grad_on_execution is None:
```

Reply:
I see! That adds up!